packages/video_player/video_player/lib/video_player.dart

```diff
@@ -575,6 +584,22 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
     await _applyPlaybackSpeed();
   }
 
+  /// Sets the caption offset of [this].
+  ///
+  /// The [offset] will be used when getting the correct caption for a specific position in [this]
+  /// The [offset] can be positive or negative.
+  ///
+  /// The values will be handled as follows:
+  /// *  0: This is the default behaviour. No offset will be applied.
+  /// * >0: The caption will have a negative offset. So you will get caption text from the past
+  /// * <0: The caption will have a positive offset. So you will get caption text from the future
+  void setCaptionOffset(Duration offset) async {
```

Comment:
Sorry, I missed this before: why is this async? (And if it really does need to be async, then the tests should be `await`ing calls to it to prevent flake).